src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/abi/x64/sysv/SysVVaList.java

```diff
@@ -121,6 +127,23 @@
         slices.add(regSaveArea);
     }
 
+    private static MemoryAddress emptyListAddress() {
+        MemorySegment ms = MemorySegment.allocateNative(LAYOUT);
+        cleaner.register(SysVVaList.class, ms::close);
```

Comment:
This will fail because of confinement

Reply:
Thanks, I fixed it by switching to Unsafe for the allocation, so that we can free from any thread.